package/MDAnalysis/topology/ITPParser.py

```diff
@@ -0,0 +1,226 @@
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+ITP topology parser
+===================
+
+Reads a GROMACS ITP_ file to build the system. The topology will
+contain atom IDs, segids, residue IDs, residue names, atom names, atom types,
+charges, chargegroups, masses (guessed if not found), and moltypes. 
+Bonds, angles, dihedrals and impropers are also read from
+the file.
+
+.. _ITP: http://manual.gromacs.org/current/reference-manual/topologies/topology-file-formats.html#molecule-itp-file
+
+Classes
+-------
+
+.. autoclass:: ITPParser
+   :members:
+   :inherited-members:
+
+"""
+from __future__ import absolute_import, division
+from six.moves import range
+
+import logging
+import functools
+import re
+from math import ceil
+import numpy as np
+
+from ..lib.util import openany
+from . import guessers
+from .base import TopologyReaderBase, squash_by, change_squash
+from ..core.topologyattrs import (
+    Atomids,
+    Atomnames,
+    Atomtypes,
+    Masses,
+    Moltypes,
+    Charges,
+    Resids,
+    Resnums,
+    Resnames,
+    Segids,
+    Bonds,
+    Angles,
+    Dihedrals,
+    Impropers,
+    AtomAttr,
+)
+from ..core.topology import Topology
+
+logger = logging.getLogger("MDAnalysis.topology.ITP")
+
+
+class Chargegroups(AtomAttr):
+    """The charge group for each Atom"""
+    attrname = 'chargegroups'
+    singular = 'chargegroup'
+
+class ITPParser(TopologyReaderBase):
+    """Read topology information from a CHARMM/NAMD/XPLOR ITP_ file.
+
+    Creates a Topology with the following Attributes:
+    - ids
+    - names
+    - types
+    - masses
+    - charges
+    - resids
+    - resnames
+    - segids
+    - moltypes
+    - bonds
+    - angles
+    - dihedrals
+    - impropers
+
+    .. _ITP: http://www.charmm.org/documentation/c35b1/struct.html
+    """
+    format = 'ITP'
+
+    def parse(self, **kwargs):
+        """Parse ITP file into Topology
+
+        Returns
+        -------
+        MDAnalysis *Topology* object
+        """
+        ids = []
+        types = []
+        resids = []
+        resnames = []
+        names = []
+        chargegroups = []
+        charges = []
+        masses = []
+
+        atom_lists = [ids, types, resids, resnames, names, chargegroups, charges, masses]
+
+        bonds = []
+        angles = []
+        dihedrals = []
+        impropers = []
+
+        segid = 'SYSTEM'
+        molnum = 1
+
+        section = None
+
+        # Open and check itp validity
+        with openany(self.filename) as itpfile:
+            header = re.compile(r'\[\s*(?P<section>\w+)\s*\]')
```

Comment:
Fwiw I think usually regex is slower than just checking for a left bracket and it looks like this is being called on all lines. I guess itps are probably gonna be small though

Reply:
Changed it anyway, and updated CHANGELOG